src/main/java/com/google/idea/perf/CallTreeBuilder.kt

```diff
@@ -94,9 +104,11 @@ class CallTreeBuilder(
             """.trimIndent()
         }
 
-        val now = clock.sample()
-        child.wallTime += now - child.continuedWallTime
-        child.maxWallTime = child.maxWallTime.coerceAtLeast(now - child.startWallTime)
+        if ((child.tracepointFlags and TracepointFlags.TRACE_WALL_TIME) != 0) {
```

Comment:
I wonder if we should also check `tracepointFlags` in `buildAndReset()` so that we don't accumulate `wallTime` for nodes that are currently on the stack when `buildAndReset()` is called.

Reply:
I think it dpeends on when you want the concurrent tracepoint modification to go into effect. In the PR as of now, a tracepoint change starts applying itself on next subsequent calls to the targeted method. However, for consistency's sake, I agree that it makes more sense to check `tracepointFlags` in `buildAndReset()`.